macros/src/main/scala/io/udash/macros/PropertyMacros.scala

```diff
@@ -314,32 +321,36 @@ class PropertyMacros(val ctx: blackbox.Context) extends AbstractMacroCommons(ctx
 
   def reifyRoSubSeq[A: c.WeakTypeTag, B: c.WeakTypeTag](f: c.Tree)(ev: c.Tree): c.Tree = {
     val resultType = weakTypeOf[B]
-    q"${reifySubProperty[A, B](f)}.asInstanceOf[$ReadableSeqPropertyCls[$resultType, $CastableReadablePropertyCls[$resultType]]]"
+    q"${reifySubProperty[A, B](f, subSeqValidation)}.asInstanceOf[$ReadableSeqPropertyCls[$resultType, $CastableReadablePropertyCls[$resultType]]]"
   }
 
-  private def reifySubProperty[A: c.WeakTypeTag, B: c.WeakTypeTag](f: c.Tree): c.Tree = {
+  private def reifySubProperty[A: c.WeakTypeTag, B: c.WeakTypeTag](f: c.Tree, leafValidation: Type => Unit = _ => ()): c.Tree = {
     val model = c.prefix
     val modelPath = getModelPath(f)
 
-    def checkIfIsValidPath(tree: Tree): Boolean = tree match {
-      case Select(next@Ident(_), t) if isValidSubproperty(next.tpe, t) =>
-        true
-      case Select(next, t) if hasModelPropertyCreator(next.tpe.widen) && isValidSubproperty(next.tpe, t) =>
-        checkIfIsValidPath(next)
-      case Select(next, t) =>
-        c.abort(c.enclosingPosition,
-          s"""
-             |The path must consist of ModelProperties and only leaf can be a Property, ModelProperty or SeqProperty.
-             | * ${next.tpe.widen} ${if (hasModelPropertyCreator(next.tpe.widen)) "is" else "is NOT"} a ModelProperty
-             | * $t ${if (isValidSubproperty(next.tpe, t)) "is" else "is NOT"} a valid subproperty (abstract val/def for trait based model or constructor element for (case) class based model)
-             |
+    def checkIfIsValidPath(tree: Tree, validation: Type => Unit = _ => ()): Boolean = {
+      tree match {
+        case Select(next@Ident(_), t) if isValidSubproperty(next.tpe, t) =>
+          validation(next.tpe.member(t).typeSignature.finalResultType)
```

Comment:
`validation` -> `abortIfInvalid`? It looks like ignored check and nothing suggests it has a side-effect.

Reply:
Validation is usually side-effectful (see property validation :rofl:). I'd say the return type `Unit` says a lot as well. This does not always have to abort from the perspective of this method - it may just be some logging etc. I'll switch to `onLeafType`